### PR TITLE
validating that layouts_container's name equals id

### PR DESCRIPTION
**Pull request by merit-maita**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/35081

## Description
Added a validation that layoutscontainer's id and name are matching.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/common/constants.py [hunk 1]
```diff
@@ -459,7 +459,8 @@ class FileType(Enum):
 ID_IN_ROOT = [  # entities in which 'id' key is in the root
     'playbook',
     'dashboard',
-    'incident_type'
+    'incident_type',
+    'layoutscontainer'
```
Comment (yaakovi) [inline demisto_sdk/commands/common/constants.py:463]:
```suggestion
    'layoutscontainer',
```

File: demisto_sdk/commands/common/hook_validations/layout.py [hunk 2]
```diff
@@ -237,3 +250,6 @@ def is_incident_field_exist(self, id_set_file, is_circle) -> bool:
             if self.handle_error(error_message, error_code, file_path=self.file_path):
                 return False
         return True
+
+    def is_id_equals_name(self) -> bool:
+        return True
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/layout.py:255]: add a comment explaining the hard-coded value (there is no name in legacy layouts)

File: demisto_sdk/commands/common/hook_validations/layout.py [hunk 3]
```diff
@@ -155,6 +160,14 @@ def is_incident_field_exist(self, id_set_file, is_circle) -> bool:
                 return False
         return True
 
+    def is_id_equals_name(self):
+        """Check whether the playbook ID is equal to its name.
+
+        Returns:
+            bool. Whether the file id equals to its name
+        """
+        return super(LayoutsContainerValidator, self)._is_id_equals_name('layoutscontainer')
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/layout.py:169]: there is no need to pass parameters to the super function in python3:
[PEP](https://www.python.org/dev/peps/pep-3135/#id4)

File: demisto_sdk/commands/common/tests/layout_test.py [hunk 4]
```diff
@@ -78,3 +78,24 @@ def test_layout_container_is_incident_field_exist_in_content(self, repo, layout_
...
         validator = LayoutsContainerValidator(structure)
         assert validator.is_incident_field_exist(id_set_json, is_circle) == expected_result
+
+    IS_MATCHING_NAME_ID_INPUT = [
+        ({"id": "name", "name": "name"}, True),
+        ({"id": "id_field", "name": "name_field"}, False)
+    ]
+
+    @pytest.mark.parametrize("layout_container, result", IS_MATCHING_NAME_ID_INPUT)
+    def test_is_name_id_equal(self, repo, layout_container, result):
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tests/layout_test.py:88]: add a test for the legacy layout too

File: demisto_sdk/commands/common/hook_validations/layout.py (hunk 2 above)
Comment (merit-maita) [reply demisto_sdk/commands/common/hook_validations/layout.py:255]: i dont quiet understated what you mean

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,6 @@
 # Changelog
 * Updated the **demisto-sdk validate** command to check that the 'additionalinfo' field only contains the expected value for feed required parameters and not equal to it.
+* Added a validation that layoutscontainer's id and name are matching
```
Comment (Bargenish) [inline CHANGELOG.md:3]: Let's add the changes to the format command as well.

File: demisto_sdk/commands/format/update_layout.py [hunk 6]
```diff
@@ -104,6 +104,7 @@ def layoutscontainer__run_format(self):
         self.set_fromVersion(from_version=VERSION_6_0_0)
         self.set_group_field()
         self.layoutscontainer__set_output_path()
+        self.update_id()
```
Comment (Bargenish) [inline demisto_sdk/commands/format/update_layout.py:107]: Please add a unit-test.

**Review by Bargenish — CHANGES_REQUESTED** [review]

**Review by yaakovi — APPROVED** [review]

**Review by Bargenish — APPROVED** [review]

**Pull request merged**